scripts/typescript_checks.py

```diff
@@ -82,12 +835,91 @@ def compile_and_check_typescript(config_path: str) -> None:
-        sys.exit(1)
+    if config_path == STRICT_TSCONFIG_FILEPATH:
+        # Generate file names from the error messages.
+        errors = [x.strip() for x in error_messages]
+        # Remove the empty lines and error explanation lines.
+        prefixes = ('core', 'extension', 'typings')
+        errors = [x for x in errors if x.startswith(prefixes)]
+        # Remove error explanation lines.
+        errors = [x.split('(', 1)[0] for x in errors]
+        # Remove the duplicate occurrences of the file names.
+        errors = list(dict.fromkeys(errors))
+        files_with_errors = sorted(errors)
+
+        # List of missing files that are neither strictly typed nor present in
+        # NOT_FULLY_TYPE_STRICT_TSCONFIG_FILEPATH.
+        files_not_type_strict = []
+        for filename in files_with_errors:
+            if filename not in NOT_FULLY_TYPE_STRICT_TSCONFIG_FILEPATH:
+                files_not_type_strict.append(filename)
+
+        # Add "typings" folder to get global imports while compiling.
+        files_not_type_strict.append('typings')
+
+        # Update "include" field of tsconfig-strict.json with files that are
+        # neither strict typed nor present in
+        # NOT_FULLY_TYPE_STRICT_TSCONFIG_FILEPATH.
+        # Example: List "files_not_type_strict".
+        file_name = os.path.join(os.getcwd(), 'tsconfig-strict.json')
+        with open(file_name, 'r', encoding='utf-8') as f:
+            tsconfig_strict_json_dict = yaml.safe_load(f)
+            tsconfig_strict_json_dict['include'] = files_not_type_strict
+
+        tsconfig_strict_json_dict = (
+            json.dumps(tsconfig_strict_json_dict, indent=2, sort_keys=True))
+
+        with open(file_name, 'w', encoding='utf-8') as f:
+            f.write(tsconfig_strict_json_dict + '\n')
+
+        # Compile tsconfig-strict.json with updated "include" property.
+        os.environ['PATH'] = '%s/bin:' % common.NODE_PATH + os.environ['PATH']
+        validate_compiled_js_dir()
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
+
+        cmd = ['./node_modules/typescript/bin/tsc', '--project', config_path]
+        process = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, encoding='utf-8')
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
+
+        # The value of `process.stdout` should not be None since we passed
+        # the `stdout=subprocess.PIPE` argument to `Popen`.
+        assert process.stdout is not None
+        error_messages = list(iter(process.stdout.readline, ''))
+
+        # Update tsconfig-strict.json and set to its intial "include" state
+        # example "include": ["core", "extensions", "typings"].
+        with open(file_name, 'r', encoding='utf-8') as f:
+            tsconfig_strict_json_dict = yaml.safe_load(f)
+            tsconfig_strict_json_dict['include'] = (
+                ['core', 'extensions', 'typings'])
```

Comment:
If the checks get aborted part-way through, we should make sure that `tsconfig-strict.json` is restored. Ideally though, it'd be great if we could avoid modifying config files dynamically like this. Is that possible?

Reply:
Also this list is already defined in `prefixes` variable.

Reply:
Where are you restoring the include property if the check gets aborted? I would expect to see this handled using a context manager (preferably) or using atexit (less preferably), but I don't see that in your code so far

Reply:
Done. Please verify, using context manager for first time.